More optional params before required ones

```
phplint 2.0.2 by overtrue and contributors.

No config file loaded.

.......................................    40 / 310 (12%)
........................................    80 / 310 (25%)
........................................   120 / 310 (38%)
..................................W.....   160 / 310 (51%)
........................................   200 / 310 (64%)
................................W.......   240 / 310 (77%)
........................................   280 / 310 (90%)
...............................

Time: 8 secs	Memory: 4.0 MiB	Cache: No

FAILURES!
Files: 310, Failures: 2

There was 2 errors:
1. /home/runner/work/SMF2.1/SMF2.1/Sources/Subs-Package.php:1593
    1590|  * @param string $the_version
    1591|  * @return string|bool Highest install value string or false
    1592|  */
  > 1593| function matchHighestPackageVersion($versions, $reset = false, $the_version)
    1594| {
    1595| 	static $near_version = 0;
    1596| 
  Required parameter $the_version follows optional parameter $reset in line 1593
2. /home/runner/work/SMF2.1/SMF2.1/Sources/ManageBans.php:936
    933|  *
    934|  * @return array|bool An array with the triggers if there were errors or false on success
    935|  */
  > 936| function saveTriggers($suggestions = array(), $ban_group, $member = 0, $ban_id = 0)
    937| {
    938| 	global $context;
    939| 
  Required parameter $ban_group follows optional parameter $suggestions in line 936
Error: Process completed with exit code 1. 
```

I'll fix these later if no one does it first.

--- a/Sources/ManageBans.php
+++ b/Sources/ManageBans.php
@@ -933,7 +933,7 @@ function banEdit2()
  *
  * @return array|bool An array with the triggers if there were errors or false on success
  */
-function saveTriggers($suggestions = array(), $ban_group, $member = 0, $ban_id = 0)
+function saveTriggers(array $suggestions, $ban_group, $member = 0, $ban_id = 0)
 {
 	global $context;
 

--- a/Sources/Subs-Package.php
+++ b/Sources/Subs-Package.php
@@ -1580,17 +1580,18 @@ function parsePackageInfo(&$packageXML, $testing_only = true, $method = 'install
 }
 
 /**
- * Checks if version matches any of the versions in versions.
+ * Checks if version matches any of the versions in `$versions`.
+ *
  * - supports comma separated version numbers, with or without whitespace.
  * - supports lower and upper bounds. (1.0-1.2)
  * - returns true if the version matched.
  *
- * @param string $versions The SMF versions
+ * @param string[] $versions The SMF versions
  * @param boolean $reset Whether to reset $near_version
  * @param string $the_version
  * @return string|bool Highest install value string or false
  */
-function matchHighestPackageVersion($versions, $reset = false, $the_version)
+function matchHighestPackageVersion(array $versions, $reset, $the_version)
 {
 	static $near_version = 0;
 

